contract_execution_result.js

```diff
@@ -5,13 +5,15 @@ const {AssetProof} = require('./asset_proof.js');
  */
 class ContractExecutionResult {
   /**
-   * @param {Object} result
-   * @param {Array} proofs
-   * @param {Array} [auditorProofs = []]
+   * @param {string} contractResult
+   * @param {string} functionResult
+   * @param {AssetProof[]} ledgerProofs
+   * @param {AssetProof[]} auditorProofs
    */
-  constructor(result, proofs, auditorProofs = []) {
-    this.result = result;
-    this.proofs = proofs;
+  constructor(contractResult, functionResult, ledgerProofs, auditorProofs) {
+    this.contractResult = contractResult;
+    this.functionResult = functionResult;
+    this.ledgerProofs = ledgerProofs;
```

Comment:
Please don't worry about the change of the constructor. the upper layer doesn't use it. The instances will be created by another function.